ultravox/data/datasets.py

```diff
@@ -289,134 +300,116 @@ class VoiceDataset(SizedIterableDataset):
     Wraps a Hugging Face dataset or MDS-formatted dataset from GCP.
     """
 
-    BASE_AUDIO_COLUMNS = ["audio"]
-
     def __init__(self, args: VoiceDatasetArgs) -> None:
         super().__init__()
         self._args = args
-        self._session: Optional[requests.Session] = None
         self._rng = np.random.default_rng(self._args.shuffle_seed)
-        self._weight = 1.0  # the default weight for the dataset
 
-    def _init_dataset(self, dataset: data.Dataset, estimated_length: int = 1) -> None:
+    def _init_dataset(self, dataset: data.Dataset, num_samples: int) -> None:
         self._dataset = dataset
-        # Only required when using epochs when training dataset.
-        self._estimated_length = estimated_length
+        self._length = num_samples
 
-    @property
-    def weight(self) -> float:
-        return self._weight
+    def __len__(self):
+        return self._length
 
-    def _load_audio_dataset(
+    def _load_hf_dataset(
         self,
         path: str,
         name: Optional[str] = None,
         *,
         split: Optional[str] = None,
-        shuffle: Optional[bool] = None,
         streaming: bool = True,
+        audio_field: Optional[str] = None,
```

Comment:
Yeah, the real challenge here is figuring out which audio to load in `_get_audio` since we don't have a mechanism to select that on a sample-by-sample basis

Reply:
Accordingly, will plan to leave as-is for now.